docs/sorting-algorithms/heap-sort.mdx

```diff
@@ -0,0 +1,233 @@
+
+* Swap the root of the MaxHeap with the last element of the MaxHeap 
+
+* Reduce the size of MaxHeap by 1 from the last and call Heapify on the root.
+
+* Perform step 2 and step 3 
+
+Heap Sort for Sorting in Decreasing Order:
+
+* First we will convert array into MinHeap by calling Heapify function on the range `[(n-2)/2, 0]`. `(n-2)/2` is the last internal node
+
+* Swap the root of the MinHeap with the last element of the MinHeap 
+
+* Reduce the size of MinHeap by 1 from the last and call Heapify on the root.
+
+* Perform step 2 and step 3 
+
+## Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include <iostream>
+using namespace std;
+
+void Heapify(int arr[],int idx,int size){
+    int l=2*idx+1;
+    int r=2*idx+2;
+    int maxx=idx;
+    
+    if(l<size && arr[l]>arr[maxx])maxx=l;
+    if(r<size && arr[r]>arr[maxx])maxx=r;
+
+    swap(arr[maxx],arr[idx]);
+    if(maxx!=idx)
+    {
+        Heapify(arr,maxx,size);
+    }
+}
+
+int main()
+{
+    cout<<"Enter elements to be sorted :";
+    int n;
+    cin>>n;
+
+    int arr[n];
+    for(int i=0;i<n;i++)cin>>arr[i];
+
+    for(int i=(n-2)/2;i>=0;i--)          
+    {
+       Heapify(arr,i,n); 
+    }
+
+    for(int i=n-1;i>=1;i--)
+    {
+        swap(arr[0],arr[i]);
+        Heapify(arr,0,i);
+    }
+    
+    for(int i=0;i<n;i++)
+    {
+        cout<<arr[i]<<" ";
+    }
+    return 0;
+}
+```
+</TabItem>
+
+<TabItem value="py">
+
+```python   
+def heapify(arr, length, index):
+    # base case
+    # we will call this function until the largest number is the index...
+    largest_num_index = index
+    left_index = (index * 2) + 1
+    right_index = (index * 2) + 2
+ 
+    if(left_index < length and arr[index] < arr[left_index]):
+        largest_num_index = left_index
+ 
+    if(right_index < length and arr[largest_num_index] < arr[right_index]):
+        largest_num_index = right_index
+ 
+    # if index is not the largest, make it the largest!
+    # and run it again!
+    if(largest_num_index != index):
+        arr[index], arr[largest_num_index] = arr[largest_num_index], arr[index]
+        heapify(arr, length, largest_num_index)
+ 
+ 
+def heap_sort(arr):
+    # need array length to create indices
+    length = len(arr)
+ 
+    for index in range(length, -1, -1):
+        # ask about functions modifying arrays without return value
+        # build max heap
+        heapify(arr, length, index)
+ 
+    # for each sorted heap, swap the root and the last number
+    for index in range(length - 1, 0, -1):
+        arr[index], arr[0] = arr[0], arr[index]
+        # then call heapify again with the new array
+        heapify(arr, index, 0)
+ 
+ 
+# Taking Elements to be Sorted
+
+data = []
+n = int(input("Enter the Size of Array : "))
```

Comment:
The input statements of Python and C++ don't match. Kindly fix them. 

We want the I/O statement to be same for both the languages.

Reply:
done

Reply:
done